### ISSUE TITLE
Update 'flutter screenshot' to fail better if a device doesn't support screenshots

### ISSUE THREAD

**Issue by liyuqian** [flutter/flutter#35490]
I'm using Flutter master branch version 1a374d820de32d359c06a8ecf0e1348e3ce69a5a, and the following error happened although the `flutter device` does show a connected Moto G4 device:

```
liyuqian@liyuqian:~/Downloads$ flutter devices
3 connected devices:

Moto G 4 • ZY2245H6KS • android-arm    • Android 7.0 (API 24)
Linux    • Linux      • linux-x64      • Linux
Chrome   • chrome     • web-javascript • Google Chrome 75.0.3770.100 

liyuqian@liyuqian:~/Downloads$ flutter screenshot
No supported devices connected.
Must have a connected device
liyuqian@liyuqian:~/Downloads$ flutter screenshot --type=skia --observatory-uri=http://127.0.0.1:35603/atA841bQWXA=/
No supported devices connected.
Must have a connected device
```

The app that I ran is
```
liyuqian@liyuqian:~/flutter/flutter/dev/benchmarks/macrobenchmarks$ flutter run --profile
Initializing gradle...                                              0.7s
Resolving dependencies...                                           1.3s
Launching lib/main.dart on Moto G 4 in profile mode...
Running Gradle task 'assembleProfile'...                                
Running Gradle task 'assembleProfile'... Done                       1.2s
Built build/app/outputs/apk/profile/app-profile.apk (8.0MB).

An Observatory debugger and profiler on Moto G 4 is available at http://127.0.0.1:35603/atA841bQWXA=/
For a more detailed help message, press "h". To quit, press "q".

```

This blocks fixing a P0 bug for a Google's internal client.

**Comment by liyuqian** [comment]
CC @jonahwilliams

**Comment by jonahwilliams** [comment]
Try `flutter screenshot -d ZY2245H6KS` ?

**Comment by liyuqian** [comment]
Thanks @jonahwilliams ! That fixes the problem. Shall we add `-d` to the `flutter help screenshot`, or maybe set the default device to be an Android/iOS device (I guess the error happened because of web and desktop devices)?

**Comment by HansMuller** [comment]
I've run into this as well. It feels like a regression. It hasn't be necessary to specify a device before. For years now.

**Comment by jonahwilliams** [comment]
I'm not so sure, but this command has changed very little besides an update to the observatory format. For now we should document that it requires a device flag.

it's possible that something else we changed regressed this, but that should be expected given that it has almost no tests

**Comment by liyuqian** [comment]
My suspicion is on our recent desktop and web changes. Previously, there's only one Android/iOS device to choose, but now the flutter tool can see Linux and Chrome as devices. If it chooses Linux or Chrome, then the screenshot would fail I guess.

**Comment by jonahwilliams** [comment]
> Previously, there's only one Android/iOS 

Adding web/desktop/fuchsia targets didn't help, but it seems like this would have always happened had there been multiple devices connected: https://github.com/flutter/flutter/blob/758009ba7010feda282f26fd6537b5aa79394271/packages/flutter_tools/lib/src/runner/flutter_command.dart#L534

**Comment by liyuqian** [comment]
Yes, I think I mostly only have one device, and I'll add `-d` if I have multiple devices (other than Chrome/Linux).

However, the code you referenced will print a list of devices if there are more than one. But that doesn't to be the case in the error message that I saw.

**Comment by jonahwilliams** [comment]
We should update the command so that it only runs on devices that support screenshots.

**Comment by zanderso** [comment]
@jonahwilliams What is the status of this issue?

**Comment by jonahwilliams** [comment]
I'm not actively working on it. There was no change in behavior that justified a regression, but the command could be smarter in general.
